noxfile.py

```diff
@@ -70,7 +70,14 @@ def lint_setup_py(session):
 
 def default(session):
     # Install all test dependencies, then install this package in-place.
-    session.install("mock", "pytest", "pytest-cov")
+    session.install(
+        "mock",
+        "pytest",
+        "pytest-cov",
+        "opentelemetry-api",
+        "opentelemetry-sdk",
+        "opentelemetry-instrumentation",
```

Comment:
This change will get wiped out on the next code regen, let's add dependencies through the `synth.py` file instead to make them persistent.

Reply:
Adding these as external dependencies may not work since there is not a synthtool option to add external unit test dependencies, although there is an option for external system test dependencies. Should these tests be added as system tests?

Reply:
It's not ideal, but we can circumvent this limitation and add additional dependencies through the `synth.py` file. That way the changes to `noxfile.py` will not get lost on the next code regen.

Applying the following patch should do the trick:

```patch
diff --git noxfile.py noxfile.py
index 3af3d67..20b5ad2 100644
--- noxfile.py
+++ noxfile.py
@@ -70,14 +70,12 @@ def lint_setup_py(session):
 
 def default(session):
     # Install all test dependencies, then install this package in-place.
+
+    session.install("mock", "pytest", "pytest-cov")
     session.install(
-        "mock",
-        "pytest",
-        "pytest-cov",
-        "opentelemetry-api",
-        "opentelemetry-sdk",
-        "opentelemetry-instrumentation",
+        "opentelemetry-api", "opentelemetry-sdk", "opentelemetry-instrumentation",
     )
+
     session.install("-e", ".")
 
     # Run py.test against the unit tests.
diff --git synth.py synth.py
index 0e2c96e..10edae8 100644
--- synth.py
+++ synth.py
@@ -279,4 +279,18 @@ s.move(templated_files)
 # ----------------------------------------------------------------------------
 python.py_samples()
 
+# ----------------------------------------------------------------------------
+# Additional unit test dependencies
+# ----------------------------------------------------------------------------
+s.replace(
+    "noxfile.py",
+    r'session\.install\("mock", "pytest", "pytest-cov"\)',
+    """
+    \g<0>
+    session.install(
+        "opentelemetry-api", "opentelemetry-sdk", "opentelemetry-instrumentation",
+    )
+    """,
+)
+
 s.shell.run(["nox", "-s", "blacken"], hide_output=False)
```